### PR TITLE
S3 DB tests

### PR DESCRIPTION
**Pull request by DarkPurple141**
Got a bit lost in trying to refactor the db and came out with this junk 👎

### CODE REVIEW

File: backend/src/models/db/index.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+
+const createDB = require('./tables').createDB
+const devInitDB = require('./tables').devInitDB
+const insertDB = require('./tables').insertDB
+
+const DB_NAME = process.env === 'test' ? ':memory:' : ':memory:'
+
+class DB {
```
Comment (NunoDasNeves) [inline backend/src/models/db/index.js]: wow a class!

File: backend/tests/db/dbinstantiation.js [hunk 2]
```diff
@@ -1,8 +1,61 @@
 const assert = require('assert')
-const db = require('../../src/models/db')
+const db = require('../../src/models/db')._db
 
 describe('dev db starts', () => {
     it('compiles', function () {
         assert(db)
     })
+
+    /* These don't work */
```
Comment (NunoDasNeves) [inline backend/tests/db/dbinstantiation.js]: plz fix

File: frontend/src/components/AppFooter.vue [hunk 3]
```diff
@@ -1,6 +1,6 @@
 <template>
     <footer id="footer">
-        <h2>Copyright &copy; SmartCourse 2018</h2>
+        <p>Copyright &copy; SmartCourse 2018</p>
```
Comment (NunoDasNeves) [inline frontend/src/components/AppFooter.vue]: thank god

File: frontend/src/components/AppFooter.vue [hunk 4]
```diff
@@ -10,8 +10,13 @@ footer {
     color: white;
     width: 100%;
     background-color: var(--theme);
-    padding: 0px 20px;
+    padding: 10px 0px;
     position: fixed;
```
Comment (NunoDasNeves) [inline frontend/src/components/AppFooter.vue]: can we make it not fixed too while we're at it?

File: frontend/src/components/AppFooter.vue (hunk 3 above)
Comment (DarkPurple141) [reply frontend/src/components/AppFooter.vue]: The change I deliver, but not the one you deserve @NunoDasNeves

File: backend/src/controllers/course.js [hunk 5]
```diff
@@ -26,3 +26,11 @@ exports.getCourseQuestions = function ({ params, query }, res) {
 exports.getCourseReviews = function ({ params, query }, res) {
     res.send(`<h1>Page Reviews for Course #${params.id} Page #${query.p}</h1>`)
 }
+
+exports.postQuestion = function ({ params, body }, res) {
+    body.userID = 1
+    questionModel.postQuestion(params.id, body)
+        .then(data => res.json(data))
+        // TODO potentially more meaningful error code or something
```
Comment (PapaPwn) [inline backend/src/controllers/course.js]: **"or something"**. This is poor commenting Alex.

File: backend/src/models/db/index.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+        if (DB_NAME === ':memory:') {
+            devInitDB(this._db)
+        }
+    }
+
+    insert(table, data) {
+        return insertDB(this._db, table, data)
+    }
+
+    deleteDB () {
```
[the remark this answers is not in the document]
Comment (DarkPurple141) [reply backend/src/models/db/index.js]: Purpose will be to delete entries from the DB.

**Pull request merged**